From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryder Lee <ryder.lee@mediatek.com>
Date: Wed, 25 Mar 2026 10:17:23 -0700
Subject: [PATCH 918/957] wifi: mt76: mt7996: disable UNI_BSS_INFO_PROTECT_INFO
 for mt7996

The current MT7996 firmware causes TX failure and need further
investigation, so it is temporarily disabled.

MT7992 and MT7990 are working normally.

Signed-off-by: Ryder Lee <ryder.lee@mediatek.com>
Link: https://patch.msgid.link/6427326eb4e8f375c63379f7a0df7e2ae9d120a4.1774458901.git.ryder.lee@mediatek.com
Signed-off-by: Felix Fietkau <nbd@nbd.name>
(cherry picked from commit cd931f9e5fa67e2f77b9726e447fd8a06aced3d3)
---
 mt7996/mcu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 6bdce1d7..c618b26b 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -1518,6 +1518,10 @@ int mt7996_mcu_set_protection(struct mt7996_phy *phy, struct mt7996_vif_link *li
 		PROT_NONGF_STA	 = BIT(7),
 	};
 
+	/* The current firmware causes TX failure. Need further investigation */
+	if (is_mt7996(&dev->mt76))
+		return 0;
+
 	skb = __mt7996_mcu_alloc_bss_req(&dev->mt76, &link->mt76,
 					 MT7996_BSS_UPDATE_MAX_SIZE);
 	if (IS_ERR(skb))
